### PR TITLE
Update criterion benchmarks for kimchi

### PR DESCRIPTION
**Pull request by kantp**
Trying to run the criterion benchmarks for kimchi, I ran into a couple of minor issues (documentation, a macOS dependent thing, and duplicated names for different benchmarks which prevented criterion from running).

With these changes, I am able to run the criterion benchmarks for kimchi.

There's also a minor refactoring to remove some code duplication.

### CODE REVIEW

File: kimchi/benches/proof_criterion.rs [hunk 1]
```diff
@@ -5,26 +5,30 @@ pub fn bench_proof_creation(c: &mut Criterion) {
...
 
-    let ctx = BenchmarkCtx::new(14);
-    group.bench_function(
-        format!("proof creation (SRS size 2^{})", ctx.srs_size()),
-        |b| b.iter(|| black_box(ctx.create_proof())),
-    );
-
-    let proof_and_public = ctx.create_proof();
+        group.bench_function(
+            format!("proof creation ({} gates)", ctx.num_gates),
```
Comment (dannywillems) [inline kimchi/benches/proof_criterion.rs:12]: Can you add the SRS size, in log2 please? We often think in terms of log2 domain size for the benchmarks. It is good to have it on the stdout.

File: kimchi/benches/proof_criterion.rs [hunk 2]
```diff
@@ -5,26 +5,30 @@ pub fn bench_proof_creation(c: &mut Criterion) {
...
-        format!("proof verification (SRS size 2^{})", ctx.srs_size()),
-        |b| b.iter(|| ctx.batch_verification(black_box(&vec![proof_and_public.clone()]))),
-    );
+
+    for size in [10, 14] {
+        let ctx = BenchmarkCtx::new(size);
+        let proof_and_public = ctx.create_proof();
+
+        group.bench_function(
+            format!("proof verification ({} gates)", ctx.num_gates),
```
Comment (dannywillems) [inline kimchi/benches/proof_criterion.rs:27]: Same comment than above.

**Review by dannywillems — CHANGES_REQUESTED** [review]
Small changes. Otherwise, LGTM.

File: kimchi/benches/proof_criterion.rs (hunk 1 above)
Comment (kantp) [reply kimchi/benches/proof_criterion.rs]: Sure!

(The reason I had added the number of gates was that the size in log2 alone was not leading to unique names, but having both solves the issue and keeps the log2 size).

**Comment by kantp** [comment]
Thanks for the quick review! I made the changes you requested.

**Review by dannywillems — APPROVED** [review]

**Comment by dannywillems** [comment]
Accepted. Running the CI and will merge it.

**Pull request merged**